### DISCUSSION TITLE
HWInfoPlugin error - value changes to NaN%

### DISCUSSION THREAD

**Discussion by UnsurpassedCheese** [Rem0o/FanControl.Releases#1904] [General]
I’ve been having a re-occurring issue over the last 4-5 days that I believe is caused by the HWInfoPlugin and a value taken from it being occasionally interpreted as NaN%

My CPU & GPU are water-cooled in the same custom loop - The Motherboard is an Asus ROG Z690 and I’m using the T-Sensor1 to measure the coolant temperature, which is exposed to HWInfo64 and then into Fan Control via HWInfoPlugin. This is the only sensor I'm exposing to the gadget from HWInfo64.

My config is fairly simple - CPU/GPU temps are in a Mix and control the pump speed, the T-Sensor1 dictates the fan speeds for the radiators. I’ve had this setup for about a 7-8 weeks now without issue.

Except every now and then (and only in the last week or so) Fan Control will encounter an exception, read the T-Sensor1 as NaN% and the fans will drop to their lowest running speed. The tray icon for the T-Sensor1 reads -2 when this errors and I’ve tried setting the fan curve to 100% below 0 degrees, but this doesn’t help since the actual reading is not a number.

I can clear this by either restarting Fan Control or switching to an older profile I have that doesn’t read the T-Sensor1 and runs the radiator fans and pump off the CPU/GPU Mix, and then back to the current one.

The T-Sensor1 is also exposed to RivaTuner Statistics Server via HWInfo64 and I’ve never noticed it error there, nor in HWInfo64. The only way to tell it has errored is to either have Fan Control window open (as it states it’s encountered an error and added to the log), or to check the task bar value shows -2 degrees for the coolant temp. Example image, Blue is Intel CPU Package temp, Green is Nvidia GPU core temp, White is T-Sensor1.

<img width="960" alt="FC" src="https://github.com/Rem0o/FanControl.Releases/assets/139238641/ae7d1c98-a622-4614-b8af-e82be50d07a7">

HWInfo64 is v750-5150, Fan Control is V162 & plugin version is V1.3

[Log file attached](https://github.com/Rem0o/FanControl.Releases/files/12017721/log.txt)

**Comment by Fr0stX76** [comment]
@UnsurpassedCheese 

I highly suspect that this is happening because the Asus EC sensor is not reporting for longer period then the refresh period of FanControl. One way to "withness" this is to trigger the collumn "latency" in HWinfo64. When mine is on the Asus EC, it sometime report over 1000ms, which I think is the threshold for FanControl to consider a sensor as "missing" (thus the NaN).

If I remember correctly when this occur, HWinfo sensor line with also go greyed (but not disapear) which also means that the sensor is gone "missing" (more like very slow at refreshing).

Sadly I do not have a solution to propose. Mine was to buy an Aquacomputing highflow NEXT, which has been amazing reading my water temp and flow...

Good luck

**Comment by UnsurpassedCheese** [comment]
@DfrostX

Thank you for the response

This does sort of make sense now – this has only seems to happen when the system is idle for a short period, or doing such light workload it’s not enough to change the coolant temps much, if at all. 

I did change the Global Polling period from the default 2000ms down to 900ms but it has happened again regardless. I’ve added the latency column to HWInfo64 and will see if I can observe this behaviour for myself.

I have also updated HWInfo64 to the latest beta version, as at this point I’ll take anything

Your recommendation of the AquaComputer flow meter has sent me down a rabbit-hole looking at their product catalogue – I’m probably going to end up getting an OCTO controller with AquaSuite and adding that into my setup if the HWInfo64 beta does not alleviate the issue.

Thanks

**Comment by Fr0stX76** [comment]
@UnsurpassedCheese 

Did some digging in the Hwinfo pluggin issues and found those:

https://github.com/Rem0o/FanControl.HWInfo/issues/36
https://github.com/Rem0o/FanControl.HWInfo/issues/23

Same issue as you (except the first one was for an NZXT device). The log error is exactly the same:

"at FanControl.HWInfo.HWInfoPlugin.Update() in C:\projects\fancontrol-hwinfo\HWInfoPlugin.cs:line XX"

From Remi: "From your log, it means the sensor goes offline from HWInfo for more than 10 seconds, and from the plugin POV that's unreliable enough to throw an error."

At this point it's either the Asus EC or the physical sensor that is too slow to report and goes offline for over 10 sec in HWinfo, which is considered missing for the plugin and FanControl, thus the error message. I would put my nmoney on the Asus EC being unreliable.

As for the Octo, I never tried it, but I read it's a good product. It is supported by HWinfo and even has a FanControl plugin.

**Reply by Rem0o** [reply]
Thanks @DfrostX
